services/motion/builtin/builtin.go

```diff
@@ -537,3 +538,73 @@ func (ms *builtIn) planMoveOnMap(
 	plan, err := motionplan.FrameStepsFromRobotPath(f.Name(), solutionMap)
 	return plan, kb, err
 }
+
+// ErrorState will provide a pose P that is the kinematic base's offset from where it would be if it were to follow a path perfectly.
+// This pose P is such that spatialmath.Compose(nominalPose, P) will yield the actual pose seen by the localizer.
+func ErrorState(
+	ctx context.Context,
+	kb kinematicbase.KinematicBase,
+	plan [][]referenceframe.Input,
+	currentNode int,
```

Comment:
The motion service will need to be keeping track of how far through its motion plan it is one way or another. This can be updated to reflect how we want to pass that information in.

One reason to pass in the full plan plus an index here, is that we need to know the intended position on the plan in addition to where we are now. For relative plans, that means that something needs to look through the plan and compute the pose based on the different sets of inputs. This would be easier to do here rather than have the motion service keep a running pose and pass it in.